text/0033-logging.md

```diff
@@ -0,0 +1,200 @@
+| WARN  | Used to warn about an exceptional non-blocking event occurring.                                                   |
+| INFO  | Default Log Level. It displays high level informations of events occuring in the search engine.                   |
+| DEBUG | Used for debugging and development purposes.  More verbose than INFO.                                             |
+| TRACE | Display everything happening at engine level. Not used at the moment.                                             |
+
+
+##### Log Format
+
+```
+[2021-03-02T20:33:09Z INFO actix_web::middleware::logger] 172.17.0.1:57220 "POST /indexes/indexUID/documents HTTP/1.1" 202 14 "-" "PostmanRuntime/7.26.10" 0.023529
+```
+
+###### Mandatory log format part. E.g [TIME_FORMAT LOG_LEVEL MODULE] part.
+
+- Time when the request was started to process (in rfc3339 format)
+- Log levels are  `ERROR`, `WARN`, `INFO`, `DEBUG`, `TRACE`.
+- The module part gives information about which module logs the entry.
+
+###### HTTP Call
+
+Transplant uses `actix_web::middleware::logger` to log informations on API endpoints that receive calls.
+
+Given
+```172.17.0.1:57220 "POST /indexes/indexUID/documents HTTP/1.1" 202 14 "-" "PostmanRuntime/7.26.10" 0.023529```
+
+- Peer IP address (or IP address of reverse proxy if used)
+- First line of request (Example: GET /test HTTP/1.1)
+- Response status code
+- Size of response in bytes, including HTTP headers
+- User-Agent
+- Time taken to serve the request, in seconds to 6 decimal places
+
+> In DEBUG level the Search endpoint should log the request body and response body.
+
+### V. Impact on Documentation
+
+The documentation only mention the logging behavior for the `development` env on the `MEILI_ENV` part.
+
+We should explain how to specify the log level using `RUST_LOG` environment variable and display the Log Levels table as information in a dedicated section.
+
+### VI. Impact on SDKs
+N/A
+
+## 2. Technical Aspects
+
+MeiliSearch is not consistent about logging methods. It have occurences of `println` and `eprintln` in the codebase and occurences of `error`, `warn`, `info`, `debug`, `trace` methods. It also calls `log::warn` or `log::error`.
+
+Milli and Transplant should be careful by keeping a consistent way to log informations.
+
+## 3. Future Possibilities
+
+- Store logs on filesystem (give us future possibilites of rolling strategy). We will keep an eye on https://roadmap.meilisearch.com/c/81-specify-log-path, Github issues and, Slack Community messages. Keep in mind that it is possible to send logs to files using `syslog` or `systemd` journalctl.
+- Develop API endpoint to search the logged events and configure logging policy within the system (SaaS feature in mind).
+
+## 4. Planned Changes
+
+### 0.21
+
+#### Core
+- Use a consistent method to log
+- Log output should start with the mandatory log format part.
```

Comment:
Should we implement the HTTP call format you describe? I would say yes so that we are iso with MeiliSearch v0.20.0 :)

Reply:
Yes!